source/blog/2016-12-14-talking-to-docker-daemon.html.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Talk to Docker Daemon of Atomic Host remotely
+author: trishnag
+date: 2016-12-14 14:00:00 UTC
+published: true
+comments: true
+tags: atomic host, ansible, docker
+---
+
+
+This post will describe how to use [Docker](https://www.docker.com/) daemon of Atomic host remotely.  Note that we are also going to secure the Docker daemon since we are connecting via Network which we will be doing with TLS.
+
+TLS (Transport Layer Security) provides communication security over computer network. We will create client cert and server cert to secure our Docker daemon. OpenSSL will be used to to create the cert keys for establishing TLS connection.
+
+I am using [Fedora-Atomic](https://getfedora.org/en/atomic/) host as remote and [workstation](https://getfedora.org/en/workstation/download/) as my present host.
+
+Thanks to [Chris Houseknecht](https://twitter.com/CHouseknecht) for writing an [Ansible](https://www.ansible.com/) role which creates all the certs required automatically, so that there is no need to issue openssl commands manually. Here is the Ansible role repository: [role-secure-docker-daemon](https://github.com/ansible/role-secure-docker-daemon). Clone it to your present working host.
+
+**The steps are as follows:**
+
+
+```
+$ mkdir secure-docker-daemon
+$ cd secure-docker-daemon
+$ mkdir local remote
+```
+
+First setup the local directory
+```
+$ cd local
+$ touch ansible.cfg inventory localsetup.yml
+```
+
+**ansible.cfg** in **local**:
+
+```
+$ vim ansible.cfg
+[defaults]
+inventory=inventory
+```
+
+**inventory** in **local**:
+
+```
+$ vim inventory
+[local]
+127.0.0.1
```

Comment:
@jberkus If you take a look at https://gist.github.com/trishnaguha/4461615d562e10e4bd72a6df5f45ae9e#file-gistfile1-txt-L11
this makes sure we are not ssh-ing to our localhost.